docs/custom_registry_login.md

```diff
@@ -0,0 +1,218 @@
+   --repository value          the run's repository
+   --branch value              the git branch
+   --triggered-by value        describes what the run was triggered by
+   --git-tag value             the git tag that triggered the run
+   --registry value, -r value  the fully qualified name of the registry
+   --set value                 set values on the command line (use --set multiple times or use commas: key1=val1,key2=val2)
+   --az-cloud-name value       the name of azure environment
+```
+
+Build and Push the acr-builder docker image to a docker hub or azure container registry. Here for simplicity , let's us push to docker hub:
+
+```
+docker build -t <your-dockerhub-username>/acb-exec .
+docker push <your-dockerhub-username>/acb-exec
+```
+
+### Run **acb** on Azure VM with Managed Service Identity enabled.
+
+We will create a Azure Linux VM with Identity enabled. To create azure VM, open either a terminal with the Azure CLI installed or check out CloudShell on Azure Portal.
+
+First, create a resource group or skip this step if you already have one:
+
+```
+az group create --name myResourceGroup --location eastus
+```
+
+### Create the User Assigned Identity
+
+Next we create a user assigned identity. Once we create and add the permissions, we will be able to use this for container instances:
+
+```
+az identity create -g myResourceGroup --name myUserIdentity
+```
+
+The output is as following
+```
+{
+  "clientId": "c72b2df0-b9d8-4ac6-9363-7c1eb06c1c86",
+  "clientSecretUrl": "<secret URL>",
+  "id": "<resourceID of identity>",
+  "location": "eastus",
+  "name": "myuseridentity",
+  "principalId": "05109190-7167-4e14-8363-95bcf8b17992",
+  "resourceGroup": "myresourcegroup",
+  "tags": {},
+  "tenantId": "72f988bf-86f1-41af-91ab-2d7cd011db47",
+  "type": "Microsoft.ManagedIdentity/userAssignedIdentities"
+}
+```
+
+lets add the important information from this output to some environment variables.
+
+```
+CLIENT_ID="<clientId>"
+PRINCIPAL_ID="<principalId>"
+ID="<id>"
+```
+
+### Create a Key Vault and Set the Permissions
+Use the following command to create key vault:
+
+```
+az keyvault create -g myResourceGroup --name myacbvault
+```
+
+We can quickly add a secret to the new vault:
+
+```
+ az keyvault secret set --name username --value "registry1_username" --vault-name myacbvault
+ az keyvault secret set --name password --value "registry1_password" --vault-name myacbvault
+```
+
+Now, we can give our identity access to the Key Vault:
+
+```
+az keyvault set-policy -n myacbvault --object-id $PRINCIPAL_ID -g myResourceGroup --secret-permissions get
+```
+The above command uses the environment variable we set to give our identity “get” permission for secrets in the Key Vault
+
+### Create Azure VM with identity
+
+```
+az vm create -n testacbvm -g myResourceGroup --image UbuntuLTS --assign-identity $ID
+```
+
+### SSH to VM and set up with docker and acb image
+
+```
+sudo -s
```

Comment:
As a best practice try not dropping to root and just run `sudo apt-get install docker-ce -y`

Reply:
I wanted to be consistent with some of our other docs:
https://github.com/Azure/acr-builder/blob/38f06e5acb94854c74f6fb03f7958dc711dfbd90/docs/task-with-secrets.md#ssh-to-vm-and-set-up-with-docker-and-acb-image

Reply:
but you're right, I updated it.